tests/requirements.txt

```diff
@@ -0,0 +1,6 @@
+pytest==8.3.3
```

Comment:
Are these really "requirements" or just "all the latest versions at the current moment"? I think, we should try to specify minimum required versions and not bump them unless we truly need some feature from new version. "~=" operator might be a better choice here, i.e.
```
pytest~=8.3
jinja2~=3.1
...
```

Reply:
I don't like this because it could cause instability in the workflow. It's better to have certain versions and update them when needed.

Reply:
If stay within SemVer-compatible versions it should generally be OK. Somehow Pearl tests didn't require versions of all the dependencies with patch level precision.

Reply:
You know, I use certain versions just to prevent them from changing.  I don't stick to any of the versions. Everything will work without versions. It is just for stability's sake.

Reply:
Do you really think there is a significant gain in stability when using exact versions vs SemVec-compatible versions? At the same time we might be missing out on performance/security enhancements, unless we regularly update this file and I don't think this chore worth the "stability gains". Also, in its current form this file doesn't really help to understand actual version requirements of the project.

Reply:
I plan to update these versions only when adding modifying tests if necessary. The best way is to test a new version before publishing. This is Python. Here, SemVec versions do not always reflect the nature of changes. I repeat I do stick to certain versions; it's just for stability.

Reply:
I think, SemVer provided stability should be good enough for the project of this size without the need for manual process.

Reply:
I think we should keep it as it is. I plan to change it only after local testing. And I don't plan to create PRs to update it.

Reply:
I don't see what is the purpose of this. Let's first see if those stability problems are actually real.

Reply:
OK, I don't see the purpose of the SemVer approach as well. Let's remove versions.

Reply:
The idea is to just specify minimally required versions. Although, there is no need to actually find the oldest versions these tests can run on, we can just declare currently used versions as minimal via ~= operator, as was shown above.

Reply:
OK. Fixed.